drivers/scheduler/rke/rke.go

```diff
@@ -1,28 +1,234 @@
+	username := "admin"
+	password := "1JVWD8juKPB8fVgK"
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+	body := strings.NewReader(fmt.Sprintf(`{"username":"%s","password":"%s"}`, username, password))
+	req, err := http.NewRequest("POST", rancherURL, body)
+	if err != nil {
+		log.Errorf("Failed to create rancher POST request:", err)
+		return err, &var1
+	}
+	req.Header.Add("Content-Type", "application/json")
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Println("Failed to send request:", err)
+		return err, &var1
+	}
+	defer resp.Body.Close()
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Println("Failed to read response:", err)
+		return err, &var1
+	}
+	response := string(respBody)
+	err = json.Unmarshal([]byte(response), &data)
+	if err != nil {
+		fmt.Println("Error:", err)
+		return err, &var1
+	}
+	var1.Endpoint = endpoint
+	var1.Token = data["token"].(string)
+	var1.AccessKey = strings.Split(data["token"].(string), ":")[0]
+	var1.SecretKey = strings.Split(data["token"].(string), ":")[1]
+	return nil, &var1
+}
+
+// GetActiveRancherClusterId returns the ID of active rancher cluster
+func (k *Rancher) GetActiveRancherClusterId() (string, error) {
+	var clusterId string
+	clusterCollection, err := k.client.Cluster.List(nil)
+	if err != nil {
+		return "", err
+	}
+	for _, cluster := range clusterCollection.Data {
+		if cluster.State == "active" {
+			clusterId = cluster.ID
+		}
+	}
+	return clusterId, nil
+}
+
+// CreateRancherProject creates new project in rancher cluster
+func (k *Rancher) CreateRancherProject(projectName string, projectDescription string) (error, *rancherClient.Project) {
+	var clusterId string
+	var newProject *rancherClient.Project
+	clusterId, err := k.GetActiveRancherClusterId()
+	if err != nil {
+		return err, newProject
+	}
+	projectRequest := &rancherClient.Project{
+		Name:        projectName,
+		Description: projectDescription,
+		ClusterID:   clusterId,
+	}
+	newProject, err = k.client.Project.Create(projectRequest)
+	if err != nil {
+		return err, newProject
+	}
+	return nil, newProject
+}
+
+// GetProjectId return the project ID
+func (k *Rancher) GetProjectId(projectName string) (error, string) {
+	var projectId string
+	projectList, err := k.client.Project.List(nil)
+	if err != nil {
+		return err, projectId
+	}
+	for _, project := range projectList.Data {
+		if project.Name == projectName {
+			projectId = project.ID
+			break
+		}
+	}
+	return nil, projectId
+}
+
+// AddNamespacesToProject adds namespace to the given project
+func (k *Rancher) AddNamespacesToProject(projectName string, nsList []string) error {
+	var projectId string
+	var err error
+	namespaceAnnotation := make(map[string]string)
+	namespaceLabel := make(map[string]string)
+	err, projectId = k.GetProjectId(projectName)
+	if err != nil {
+		return err
+	}
```

Comment:
@snigdha-px as discussed, please add comments why moving directly NS to project won't work. This will help anyone to enhance this API in future.

Reply:
done

Reply:
Added in source code